File: src/main/java/io/gaegul/ch01/puzzle/Troll.java
package io.gaegul.ch01.puzzle;

import io.gaegul.ch01.puzzle.behavior.AxeBehavior;

public class Troll extends Character {

	protected Troll() {
		super(new AxeBehavior());
	}

	@Override
	public void fight() {
		System.out.println("트롤은 ");
		this.performWeapon();
		System.out.println("으로 전투합니다.");
	}
}

File: src/main/kotlin/io/gaegul/ch01/puzzle/Troll.kt
package io.gaegul.ch01.puzzle

import io.gaegul.ch01.puzzle.behavior.AxeBehavior

class Troll: Character(
    AxeBehavior()
) {
    override fun fight() {
        print("트롤은 ")
        weaponBehavior.useWeapon()
        print("으로 전투할 수 있다.")
    }
}